vnext/local-cli/src/runWindows/utils/msbuildtools.ts

```diff
@@ -184,23 +184,22 @@ function VSWhere(
         if (version === '16.0') {
           toolsPathOutputStr = path.resolve(toolsPathOutputStr, '..');
         }
-        toolsPath = toolsPathOutputStr;
+        return toolsPathOutputStr;
       }
-    } catch (e) {
-      toolsPath = null;
-    }
-    return toolsPath;
+    } catch {}
+
+    throw new Error(`Failed to find ${requires}`);
   }
 }
 
 function getVCToolsByArch(buildArch: BuildArch): string {
-  switch (buildArch.toLowerCase()) {
+  switch (buildArch) {
```

Comment:
Using this directly means we get exhaustiveness checking. I.e. if a new value is added to BuildArch, and it isn't handled in a case here, we will get a compile-time error because we would return undefined.